KeeTrayTOTP.Tests/Menu/TrayMenuTests.cs

```diff
@@ -0,0 +1,151 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Windows.Forms;
+using FluentAssertions;
+using KeePass.App.Configuration;
+using KeePass.Forms;
+using KeePass.Plugins;
+using KeePass.UI;
+using KeePassLib;
+using KeePassLib.Keys;
+using KeePassLib.Security;
+using KeePassLib.Serialization;
+using KeeTrayTOTP.Menu;
+using KeeTrayTOTP.Tests.Extensions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+
+namespace KeeTrayTOTP.Tests.Menu
+{
+    [TestClass]
+    public class TrayMenuTests
+    {
+        private KeeTrayTOTPExt _plugin;
+        private TrayMenuItemProvider _trayMenuItemProvider;
+
+        [TestInitialize]
+        public void Initialize()
+        {
+            _plugin = CreatePluginHostMock(out var host);
+            _trayMenuItemProvider = new TrayMenuItemProvider(_plugin, host.Object);
+        }
+
+        [TestMethod]
+        public void TrayMenuItemProvider_ShouldReturnANewDropDownMenuItem()
+        {
+            var sut = _trayMenuItemProvider.ProvideMenuItem();
+
+            sut.Should().NotBeNull();
+            sut.HasDropDownItems.Should().BeTrue();
+        }
+
+        [TestMethod]
+        public void BuildMenuItemsForRootDropDown_ShouldReturnCorrectMenuItem_IfNoDatabaseIsOpened()
+        {
+            var documents = new List<PwDocument>(new[]
+            {
+                new PwDocument()
+            });
+
+            var sut = _trayMenuItemProvider.BuildMenuItemsForRootDropDown(documents).ToList();
+
+            sut.Count.Should().Be(1);
+            sut.First().Text.Should().Be(Localization.Strings.NoDatabaseIsOpened,
+                "because, there is no open database. (Keepass always provides a (new) PwDocument, even if there is no database open.");
```

Comment:
Minor thingie, the FluentAssertions library already outputs "because" in the message so this would end up like "because because". 

You could write it like this, as the test project can use newer language level than the class library that is consumed by Keepass:
```csharp
            sut.First().Text.Should().Be(Localization.Strings.NoDatabaseIsOpened,
                because: "there is no open database. (Keepass always provides a (new) PwDocument, even if there is no database open.");
```

Reply:
IntelliSense told me it will be prepended if "because" is not found.